tui/autoscroller.go

```diff
@@ -0,0 +1,89 @@
+package tui
+
+import (
+	"image"
+
+	tui "github.com/marcusolsson/tui-go"
+)
+
+var _ tui.Widget = &autoScrollArea{}
+
+// autoScrollArea is a widget to fill out space.
+type autoScrollArea struct {
+	tui.WidgetBase
+
+	Widget tui.Widget
+
+	topLeft    image.Point
+	autoscroll bool
+}
+
+// newAutoScrollArea returns a new ScrollArea.
+func newAutoScrollArea(w tui.Widget) *autoScrollArea {
+	return &autoScrollArea{
+		Widget:     w,
+		autoscroll: true,
+	}
+}
+
+// MinSizeHint returns the minimum size the widget is allowed to be.
+func (s *autoScrollArea) MinSizeHint() image.Point {
+	return image.ZP
+}
+
+// SizeHint returns the size hint of the underlying widget.
+func (s *autoScrollArea) SizeHint() image.Point {
+	return image.Pt(15, 8)
```

Comment:
magic numbers?  What does the point 15, 8 represent?

Reply:
I "forked" this code from the TUI scroll widget because I didn't like how it behaved and didn't have the API to make it better from the outside. This is in their source code 😱